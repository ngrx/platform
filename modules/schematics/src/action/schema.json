{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxAction",
  "title": "NgRx Action Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the action.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the action?"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files.",
      "default": false
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group actions file within 'actions' folder",
      "aliases": ["g"]
    },
    "api": {
      "type": "boolean",
      "default": false,
      "description": "Specifies if api success and failure actions should be generated.",
      "aliases": ["a"],
      "x-prompt": "Should we generate success and failure actions?"
    },
    "creators": {
      "type": "boolean",
      "default": true,
      "description": "Specifies whether to use creator functions for handling actions and reducers.",
      "aliases": ["c"],
      "x-prompt": "Do you want to use the create function?"
    }
  },
  "required": []
}
