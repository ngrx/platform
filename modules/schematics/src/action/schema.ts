export interface Schema {
  /**
   * The name of the component.
   */

  name: string;
  /**
   * The path to create the component.
   */

  path?: string;
  /**
   * The name of the project.
   */
  project?: string;
  /**
   * Specifies if a spec file is generated.
   */
  spec?: boolean;
  /**
   * Flag to indicate if a dir is created.
   */

  flat?: boolean;
  /**
   * Group actions file within 'actions' folder
   */

  group?: boolean;
}
