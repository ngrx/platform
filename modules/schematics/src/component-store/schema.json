{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxComponentStore",
  "title": "NgRx Component Store Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the component store.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the component store?"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component store.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "skipTests": {
      "type": "boolean",
      "default": false,
      "description": "When true, does not create test files."
    },
    "component": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring component.",
      "alias": "m",
      "subtype": "filepath",
      "x-prompt": "To which component (path) should the component store be provided in?"
    },
    "module": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath",
      "x-prompt": "To which module (path) should the component store be provided in?"
    }
  },
  "required": []
}
