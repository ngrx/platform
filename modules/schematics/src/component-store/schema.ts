export interface Schema {
  /**
   * The name of the component store.
   */
  name: string;

  /**
   * The path to create the component store.
   */
  path?: string;

  /**
   * The name of the project.
   */
  project?: string;

  /**
   * Flag to indicate if a dir is created.
   */
  flat?: boolean;

  /**
   * When true, does not create test files.
   */
  skipTests?: boolean;

  /**
   * Allows specification of the declaring component.
   */
  component?: string;

  /**
   * Allows specification of the declaring module.
   */
  module?: string;
}
