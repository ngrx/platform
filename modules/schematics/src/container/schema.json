{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxContainer",
  "title": "NgRx Container Options Schema",
  "type": "object",
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the container component?"
    },
    "inlineStyle": {
      "description": "Specifies if the style will be in the ts file.",
      "type": "boolean",
      "alias": "s"
    },
    "inlineTemplate": {
      "description": "Specifies if the template will be in the ts file.",
      "type": "boolean",
      "alias": "t"
    },
    "viewEncapsulation": {
      "description": "Specifies the view encapsulation strategy.",
      "enum": ["Emulated", "Native", "None"],
      "type": "string",
      "alias": "v"
    },
    "changeDetection": {
      "description": "Specifies the change detection strategy.",
      "enum": ["Default", "OnPush"],
      "type": "string",
      "alias": "c"
    },
    "prefix": {
      "type": "string",
      "format": "html-selector",
      "description": "The prefix to apply to generated selectors.",
      "alias": "p"
    },
    "style": {
      "description": "The file extension or preprocessor to use for style files.",
      "type": "string"
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files."
    },
    "flat": {
      "type": "boolean",
      "description": "Flag to indicate if a dir is created."
    },
    "skipImport": {
      "type": "boolean",
      "description": "Flag to skip the module import."
    },
    "selector": {
      "type": "string",
      "format": "html-selector",
      "description": "The selector to use for the component."
    },
    "module": {
      "type": "string",
      "description": "Allows specification of the declaring module.",
      "alias": "m"
    },
    "export": {
      "type": "boolean",
      "description": "Specifies if declaring module exports the component."
    },
    "state": {
      "type": "string",
      "description": "Specifies the path to the state exports."
    },
    "stateInterface": {
      "type": "string",
      "default": "State",
      "description": "Specifies the interface for the state."
    },
    "testDepth": {
      "description": "Specifies whether to create a unit test or an integration test.",
      "enum": ["unit", "integration"],
      "type": "string",
      "default": "integration"
    }
  },
  "required": []
}
