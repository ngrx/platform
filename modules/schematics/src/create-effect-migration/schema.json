{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxCreateEffectMigration",
  "title": "NgRx Effect Migration from @Effect to createEffect",
  "type": "object",
  "properties": {},
  "required": []
}
