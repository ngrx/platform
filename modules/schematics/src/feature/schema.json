{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxFeature",
  "title": "NgRx Feature State Options Schema",
  "type": "object",
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the feature.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "name": {
      "description": "The name of the feature.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the feature?"
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "module": {
      "type": "string",
      "description": "Specifies the declaring module.",
      "aliases": ["m"]
    },
    "skipTests": {
      "type": "boolean",
      "default": false,
      "description": "When true, does not create test files."
    },
    "reducers": {
      "type": "string",
      "description": "Specifies the reducers file.",
      "aliases": ["r"]
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group actions, reducers and effects within relative subfolders",
      "aliases": ["g"]
    },
    "api": {
      "type": "boolean",
      "default": false,
      "description": "Specifies if api success and failure actions, reducer, and effects should be generated as part of this feature.",
      "aliases": ["a"],
      "x-prompt": "Should we generate and wire success and failure actions?"
    },
    "creators": {
      "type": "boolean",
      "default": true,
      "description": "Specifies if the actions, reducers, and effects should be created using creator functions",
      "aliases": ["c"],
      "x-prompt": "Do you want to use the create functions?"
    }
  },
  "required": []
}
