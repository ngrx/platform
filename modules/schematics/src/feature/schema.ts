export interface Schema {
  /**
   * The name of the feature.
   */
  name: string;
  /**
   * The path to create the feature.
   */
  path?: string;
  /**
   * The name of the project.
   */
  project?: string;
  /**
   * Flag to indicate if a dir is created.
   */
  flat?: boolean;
  /**
   * Specifies if a spec file is generated.
   */
  spec?: boolean;
  /**
   * Allows specification of the declaring module.
   */

  module?: string;
  /**
   * Allows specification of the declaring reducers.
   */

  reducers?: string;
  /**
   * Specifies if this is grouped within sub folders
   */

  group?: boolean;
  /**
   * Specifies if this is grouped within a feature
   */
}
