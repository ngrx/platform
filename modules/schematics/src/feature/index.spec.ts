import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import * as path from 'path';
import { Schema as FeatureOptions } from './schema';
import {
  getTestProjectPath,
  createWorkspace,
  defaultWorkspaceOptions,
  defaultAppOptions,
} from '@ngrx/schematics-core/testing';

describe('Feature Schematic', () => {
  const schematicRunner = new SchematicTestRunner(
    '@ngrx/schematics',
    path.join(__dirname, '../../collection.json')
  );
  const defaultOptions: FeatureOptions = {
    name: 'foo',
    project: 'bar',
    module: '',
    group: false,
    entity: false,
  };

  const projectPath = getTestProjectPath();

  let appTree: UnitTestTree;

  beforeEach(async () => {
    appTree = await createWorkspace(schematicRunner, appTree);
  });

  it('should create all files of a feature', async () => {
    const options = { ...defaultOptions };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const files = tree.files;
    expect(
      files.includes(`${projectPath}/src/app/foo.actions.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.actions.spec.ts`)
    ).toBeFalsy();
    expect(
      files.includes(`${projectPath}/src/app/foo.reducer.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.reducer.spec.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.effects.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.effects.spec.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.selectors.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.selectors.spec.ts`)
    ).toBeTruthy();
    expect(files.includes(`${projectPath}/src/app/foo.model.ts`)).toBeFalsy();
  });

  it('should not create test files when skipTests is true', async () => {
    const options = { ...defaultOptions, skipTests: true };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const files = tree.files;
    expect(
      files.includes(`${projectPath}/src/app/foo.actions.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.actions.spec.ts`)
    ).toBeFalsy();
    expect(
      files.includes(`${projectPath}/src/app/foo.reducer.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.reducer.spec.ts`)
    ).toBeFalsy();
    expect(
      files.includes(`${projectPath}/src/app/foo.effects.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.effects.spec.ts`)
    ).toBeFalsy();
    expect(
      files.includes(`${projectPath}/src/app/foo.selectors.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/foo.selectors.spec.ts`)
    ).toBeFalsy();
  });

  it('should create all files of a feature to specified project if provided', async () => {
    const options = {
      ...defaultOptions,
      project: 'baz',
    };

    const specifiedProjectPath = getTestProjectPath(defaultWorkspaceOptions, {
      ...defaultAppOptions,
      name: 'baz',
    });

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const files = tree.files;
    expect(
      files.includes(`${specifiedProjectPath}/src/lib/foo.actions.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${specifiedProjectPath}/src/lib/foo.actions.spec.ts`)
    ).toBeFalsy();
    expect(
      files.includes(`${specifiedProjectPath}/src/lib/foo.reducer.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${specifiedProjectPath}/src/lib/foo.reducer.spec.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${specifiedProjectPath}/src/lib/foo.effects.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${specifiedProjectPath}/src/lib/foo.effects.spec.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${specifiedProjectPath}/src/lib/foo.selectors.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${specifiedProjectPath}/src/lib/foo.selectors.spec.ts`)
    ).toBeTruthy();
  });

  it('should create all files of a feature within grouped folders if group is set', async () => {
    const options = { ...defaultOptions, group: true };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const files = tree.files;
    expect(
      files.includes(`${projectPath}/src/app/actions/foo.actions.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/reducers/foo.reducer.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/reducers/foo.reducer.spec.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/effects/foo.effects.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/effects/foo.effects.spec.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/selectors/foo.selectors.ts`)
    ).toBeTruthy();
    expect(
      files.includes(`${projectPath}/src/app/selectors/foo.selectors.spec.ts`)
    ).toBeTruthy();
  });

  it('should respect the path provided for the feature name', async () => {
    const options = {
      ...defaultOptions,
      name: 'foo/Foo',
      group: true,
      module: 'app',
    };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const moduleFileContent = tree.readContent(
      `${projectPath}/src/app/app.module.ts`
    );

    expect(moduleFileContent).toMatchSnapshot();
  });

  it('should have all api effect if api flag enabled', async () => {
    const options = {
      ...defaultOptions,
      api: true,
    };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const fileContent = tree.readContent(
      `${projectPath}/src/app/foo.effects.ts`
    );

    expect(fileContent).toMatchSnapshot();
  });

  it('should have all api actions in reducer if api flag enabled', async () => {
    const options = {
      ...defaultOptions,
      api: true,
    };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const fileContent = tree.readContent(
      `${projectPath}/src/app/foo.reducer.ts`
    );

    expect(fileContent).toMatchSnapshot();
  });

  it('should have all api effect with prefix if api flag enabled', async () => {
    const options = {
      ...defaultOptions,
      api: true,
      prefix: 'custom',
    };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const fileContent = tree.readContent(
      `${projectPath}/src/app/foo.effects.ts`
    );

    expect(fileContent).toMatchSnapshot();
  });

  it('should have all api actions with prefix in reducer if api flag enabled', async () => {
    const options = {
      ...defaultOptions,
      api: true,
      prefix: 'custom',
    };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const fileContent = tree.readContent(
      `${projectPath}/src/app/foo.reducer.ts`
    );

    expect(fileContent).toMatchSnapshot();
  });

  it('should create all files of a feature with an entity', async () => {
    const options = { ...defaultOptions, entity: true };

    const tree = await schematicRunner.runSchematic(
      'feature',
      options,
      appTree
    );
    const paths = [
      `${projectPath}/src/app/foo.actions.ts`,
      `${projectPath}/src/app/foo.reducer.ts`,
      `${projectPath}/src/app/foo.reducer.spec.ts`,
      `${projectPath}/src/app/foo.effects.ts`,
      `${projectPath}/src/app/foo.effects.spec.ts`,
      `${projectPath}/src/app/foo.model.ts`,
    ];

    paths.forEach((path) => {
      expect(tree.files.includes(path)).toBeTruthy();
      expect(tree.readContent(path)).toMatchSnapshot();
    });
  });
});
