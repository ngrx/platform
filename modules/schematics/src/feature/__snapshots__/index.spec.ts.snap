// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature Schematic should create all files of a feature with an entity 1`] = `
"import { createActionGroup, emptyProps, props } from '@ngrx/store';
import { Update } from '@ngrx/entity';

import { Foo } from './foo.model';

export const FooActions = createActionGroup({
  source: 'Foo/API',
  events: {
    'Load Foos': props<{ foos: Foo[] }>(),
    'Add Foo': props<{ foo: Foo }>(),
    'Upsert Foo': props<{ foo: Foo }>(),
    'Add Foos': props<{ foo: Foo[] }>(),
    'Upsert Foos': props<{ foo: Foo[] }>(),
    'Update Foo': props<{ foo: Update<Foo> }>(),
    'Update Foos': props<{ foos: Update<Foo>[] }>(),
    'Delete Foo': props<{ id: string }>(),
    'Delete Foos': props<{ ids: string[] }>(),
    'Clear Foos': emptyProps(),
  }
});
"
`;

exports[`Feature Schematic should create all files of a feature with an entity 2`] = `
"import { createFeature, createReducer, on } from '@ngrx/store';
import { EntityState, EntityAdapter, createEntityAdapter } from '@ngrx/entity';
import { Foo } from './foo.model';
import { FooActions } from './foo.actions';

export const foosFeatureKey = 'foos';

export interface State extends EntityState<Foo> {
  // additional entities state properties
}

export const adapter: EntityAdapter<Foo> = createEntityAdapter<Foo>();

export const initialState: State = adapter.getInitialState({
  // additional entity state properties
});

export const reducer = createReducer(
  initialState,
  on(FooActions.addFoo,
    (state, action) => adapter.addOne(action.foo, state)
  ),
  on(FooActions.upsertFoo,
    (state, action) => adapter.upsertOne(action.foo, state)
  ),
  on(FooActions.addFoos,
    (state, action) => adapter.addMany(action.foos, state)
  ),
  on(FooActions.upsertFoos,
    (state, action) => adapter.upsertMany(action.foos, state)
  ),
  on(FooActions.updateFoo,
    (state, action) => adapter.updateOne(action.foo, state)
  ),
  on(FooActions.updateFoos,
    (state, action) => adapter.updateMany(action.foos, state)
  ),
  on(FooActions.deleteFoo,
    (state, action) => adapter.removeOne(action.id, state)
  ),
  on(FooActions.deleteFoos,
    (state, action) => adapter.removeMany(action.ids, state)
  ),
  on(FooActions.loadFoos,
    (state, action) => adapter.setAll(action.foos, state)
  ),
  on(FooActions.clearFoos,
    state => adapter.removeAll(state)
  ),
);

export const foosFeature = createFeature({
  name: foosFeatureKey,
  reducer,
  extraSelectors: ({ selectFoosState }) => ({
    ...adapter.getSelectors(selectFoosState)
  }),
});

export const {
  selectIds,
  selectEntities,
  selectAll,
  selectTotal,
} = foosFeature;
"
`;

exports[`Feature Schematic should create all files of a feature with an entity 3`] = `
"import { reducer, initialState } from './foo.reducer';

describe('Foo Reducer', () => {
  describe('unknown action', () => {
    it('should return the previous state', () => {
      const action = {} as any;

      const result = reducer(initialState, action);

      expect(result).toBe(initialState);
    });
  });
});
"
`;

exports[`Feature Schematic should create all files of a feature with an entity 4`] = `
"import { Injectable } from '@angular/core';
import { Actions, createEffect, ofType } from '@ngrx/effects';

import { concatMap } from 'rxjs/operators';
import { Observable, EMPTY } from 'rxjs';
import { FooActions } from './foo.actions';

@Injectable()
export class FooEffects {


  loadFoos$ = createEffect(() => {
    return this.actions$.pipe(

      ofType(FooActions.loadFoos),
      /** An EMPTY observable only emits completion. Replace with your own observable API request */
      concatMap(() => EMPTY as Observable<{ type: string }>)
    );
  });

  constructor(private actions$: Actions) {}
}
"
`;

exports[`Feature Schematic should create all files of a feature with an entity 5`] = `
"import { TestBed } from '@angular/core/testing';
import { provideMockActions } from '@ngrx/effects/testing';
import { Observable } from 'rxjs';

import { FooEffects } from './foo.effects';

describe('FooEffects', () => {
  let actions$: Observable<any>;
  let effects: FooEffects;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        FooEffects,
        provideMockActions(() => actions$)
      ]
    });

    effects = TestBed.inject(FooEffects);
  });

  it('should be created', () => {
    expect(effects).toBeTruthy();
  });
});
"
`;

exports[`Feature Schematic should create all files of a feature with an entity 6`] = `
"export interface Foo {
  id: string;
}
"
`;

exports[`Feature Schematic should have all api actions in reducer if api flag enabled 1`] = `
"import { createFeature, createReducer, on } from '@ngrx/store';
import { FooActions } from './foo.actions';

export const fooFeatureKey = 'foo';

export interface State {

}

export const initialState: State = {

};

export const reducer = createReducer(
  initialState,
  on(FooActions.loadFoos, state => state),
  on(FooActions.loadFoosSuccess, (state, action) => state),
  on(FooActions.loadFoosFailure, (state, action) => state),
);

export const fooFeature = createFeature({
  name: fooFeatureKey,
  reducer,
});

"
`;

exports[`Feature Schematic should have all api actions with prefix in reducer if api flag enabled 1`] = `
"import { createFeature, createReducer, on } from '@ngrx/store';
import { FooActions } from './foo.actions';

export const fooFeatureKey = 'foo';

export interface State {

}

export const initialState: State = {

};

export const reducer = createReducer(
  initialState,
  on(FooActions.customFoos, state => state),
  on(FooActions.customFoosSuccess, (state, action) => state),
  on(FooActions.customFoosFailure, (state, action) => state),
);

export const fooFeature = createFeature({
  name: fooFeatureKey,
  reducer,
});

"
`;

exports[`Feature Schematic should have all api effect if api flag enabled 1`] = `
"import { Injectable } from '@angular/core';
import { Actions, createEffect, ofType } from '@ngrx/effects';
import { catchError, map, concatMap } from 'rxjs/operators';
import { Observable, EMPTY, of } from 'rxjs';
import { FooActions } from './foo.actions';


@Injectable()
export class FooEffects {

  loadFoos$ = createEffect(() => {
    return this.actions$.pipe(

      ofType(FooActions.loadFoos),
      concatMap(() =>
        /** An EMPTY observable only emits completion. Replace with your own observable API request */
        EMPTY.pipe(
          map(data => FooActions.loadFoosSuccess({ data })),
          catchError(error => of(FooActions.loadFoosFailure({ error }))))
      )
    );
  });


  constructor(private actions$: Actions) {}
}
"
`;

exports[`Feature Schematic should have all api effect with prefix if api flag enabled 1`] = `
"import { Injectable } from '@angular/core';
import { Actions, createEffect, ofType } from '@ngrx/effects';
import { catchError, map, concatMap } from 'rxjs/operators';
import { Observable, EMPTY, of } from 'rxjs';
import { FooActions } from './foo.actions';


@Injectable()
export class FooEffects {

  customFoos$ = createEffect(() => {
    return this.actions$.pipe(

      ofType(FooActions.customFoos),
      concatMap(() =>
        /** An EMPTY observable only emits completion. Replace with your own observable API request */
        EMPTY.pipe(
          map(data => FooActions.customFoosSuccess({ data })),
          catchError(error => of(FooActions.customFoosFailure({ error }))))
      )
    );
  });


  constructor(private actions$: Actions) {}
}
"
`;

exports[`Feature Schematic should respect the path provided for the feature name 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { StoreModule } from '@ngrx/store';
import * as fromFoo from './foo/reducers/foo.reducer';
import { EffectsModule } from '@ngrx/effects';
import { FooEffects } from './foo/effects/foo.effects';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    StoreModule.forFeature(fromFoo.fooFeatureKey, fromFoo.reducer),
    EffectsModule.forFeature([FooEffects])
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;
